peachjam/adapters/judgments.py

```diff
@@ -0,0 +1,309 @@
+            f.write(r.content)
+            f.flush()
+            mimetype = magic.from_file(f.name, mime=True)
+            ext = guess_extension(mimetype) or ""
+            filename = f"{slugify(doc['title'])[:200]}{ext}"
+
+            sf, _ = SourceFile.objects.update_or_create(
+                document=created_doc,
+                defaults={
+                    "file": File(f, filename),
+                    "mimetype": mimetype,
+                },
+            )
+            sf.ensure_file_as_pdf()
+
+    def get_content_html(self, doc):
+        try:
+            url = f"{self.api_url}/judgments{doc['expression_frbr_uri']}/.html"
+            response = self.client_get(url).text
+            return response
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 404:
+                return ""
+            raise e
+
+    def delete_document(self, expression_frbr_uri):
+        url = f"{self.api_url}/judgments{expression_frbr_uri}"
+
+        try:
+            self.client_get(url)
+        except requests.exceptions.HTTPError as e:
+            if e.response.status_code == 404:
+                document = Judgment.objects.filter(
+                    expression_frbr_uri=expression_frbr_uri
+                ).first()
+                if document:
+                    document.delete()
+            else:
+                raise e
+
+
+@plugins.register("ingestor-adapter")
+class JudgmentDeleteAdapter(BaseJudgmentAdapter):
+    def check_for_updates(self, last_refreshed):
+        from peachjam.tasks import get_deleted_documents
+
+        # kick of bg tasks in batches
+        judgments_count = Judgment.objects.count()
+        log.info(f"Judgments count: {judgments_count}")
+        batch_range = self.settings.get("batch_range") or 10000
+        batch_range = int(batch_range)
+        batches = math.ceil(judgments_count / batch_range)
+        log.info(f"Batches count: {batches}")
+        for r in range(0, batches):
+            get_deleted_documents(
+                self.ingestor.pk, r * batch_range, (r + 1) * batch_range
+            )
+        return [], []
+
+    def get_deleted_documents(self, range_start, range_end):
+        from peachjam.tasks import delete_document
+
+        log.info(f"checking range {range_start} - {range_end}")
+        expression_frbr_uris = (
+            Judgment.objects.filter(court__code__in=self.court_codes)
+            .order_by("pk")
+            .values_list("expression_frbr_uri", flat=True)[range_start:range_end]
+        )
+        url = f"{self.api_url}/validate-expression-frbr-uris"
+        result = self.client_request(
+            "post", url, json={"expression_frbr_uris": list(expression_frbr_uris)}
+        ).json()
+        invalid_frbr_uris = result["invalid_expression_frbr_uris"]
+        log.info(f"found {len(invalid_frbr_uris)} documents to delete")
+
+        qs = Judgment.objects.filter(court__code__in=self.court_codes).values_list(
```

Comment:
this seems unnecessary -- why not just call delete_document with the invalid uris? If you really want to validate them to ensure the server didn't send you something back you didn't ask about, you can compare it with the submitted list, rather than going back to the db.